src/utils/reportHelpers.ts

```diff
@@ -249,40 +333,150 @@ const getRowType = (type: string) => {
   }
 }
 
-export const convertToReportRows = (coordinatorRows: IPlanningRow[], reportType: ReportType | ''): IBudgetBookSummaryTableRow[] | IStrategyTableRow[] => {
-  if (reportType === 'budgetBookSummary') {
-    let forcedToFrameHierarchy: IBudgetBookSummaryTableRow[] = [];
-    forcedToFrameHierarchy = getBudgetBookSummaryProperties(coordinatorRows);
-    getInvestmentPart(forcedToFrameHierarchy);
-    return forcedToFrameHierarchy;
-  }
-  else if (reportType === 'strategy') {
-    const forcedToFrameHierarchy: IStrategyTableRow[] = [];
-    for (const c of coordinatorRows) {
-      const convertedClass = {
-        id: c.id,
-        name: c.name,
-        parent: null,
-        children: c.children.length ? convertToReportRows(c.children, reportType) : [],
-        projects: c.projectRows.length ? convertToReportProjects(c.projectRows) : [],
-        costForecast: c.cells[0].plannedBudget,
-        type: getRowType(c.type) as ReportTableRowType
+const getExtraRows = (project: IPlanningRow) => {
+  const categories: ICategoryArray[] = [];
+  // Map category rows
+  Object.keys(Categories).forEach(key => {
+    const categoryKey = key as keyof typeof Categories;
+    categories.push({
+      children: [],
+      frameBudgets: [],
+      plannedBudgets: {},
+      plannedBudgetsForCategories: getPlannedBudgetsByCategories(project, Categories[categoryKey]),
+      id: `category-${key}-${project.id}`,
+      name: t(`option.${key}`),
+      projects: [],
+      type: "category",
```

Comment:
The key is already a string, so maybe there's no need to get it from Categories? But this kind of relates to the previous question of if the category needs to be typeof Categories. If not, then the categories could be used from the store (listsSlice).

Reply:
I need to use the `categoryKey` here because I need to pass the category to `getPlannedbudgetsByCategories` function and if I try to pass the category as a string I get a type error because that function expects the type to be  of Categories. I wrote that function to expect the category to be typeof Categories in order to have a better type check there

Reply:
The Categories enum needs to be used here also because with the slice solution it would be difficult to get the budgets for the children items